### ISSUE TITLE
Numerical issue with torch_geometric.utils.softmax

### ISSUE THREAD

**Issue by jaromiru** [rusty1s/pytorch_geometric#2013]
I suspect there is an issue with `.exp()`. Minimal working example:
```
import torch, torch_geometric

act = torch.tensor([-26.8810, -112.1200,  -92.3860,  -100.1800,  -57.9930, 
                   -134.3900, -122.0100, -122.0100,  -134.3900, -122.0100])
ind = torch.tensor([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])

probs = torch_geometric.utils.softmax(act, ind)
probs_geometric = [probs[0:5], probs[5:10]]

# for reference
probs_torch = [
        act[0:5].softmax(0),
        act[5:10].softmax(0)
]

print(probs_geometric)
print(probs_torch)
```

Outputs:

[tensor([1.0000e+00, 9.5758e-38, 3.5607e-29, 1.4677e-32, 3.0777e-14]), tensor(**[0.0000e+00, 4.8527e-26, 4.8527e-26, 0.0000e+00, 4.8527e-26]**)]
[tensor([1.0000e+00, 9.5758e-38, 3.5607e-29, 1.4677e-32, 3.0777e-14]), tensor([1.4006e-06, 3.3333e-01, 3.3333e-01, 1.4006e-06, 3.3333e-01])]

**Comment by rusty1s** [comment]
Yeah, you are right. This got introduced by this PR: https://github.com/rusty1s/pytorch_geometric/pull/1707. The numerical instability is caused by the high variance of values between different indices, so that global normalization is not sufficient to accurately compute the softmax output.

I reverted the changes and the results are now as expected again. Thanks for letting me know.

**Comment by jaromiru** [comment]
Indeed, it was caused by the global _max_ normalization. It now works as expected. Btw. you're fast! :-)
